### PR TITLE
ci: check for failed services after boot

### PR DESCRIPTION
**Pull request by mrc0mmand**
This should, hopefully, catch issues like #21671
automagically.

---

There is a couple of *iffy* parts:

1) OpenSUSE workaround in `mkosi.postinst`
This is, unfortunately, necessary, until https://github.com/openSUSE/suse-module-tools/commit/158643414ddb8d8208016a5f03a4484d58944d7a# lands in OpenSUSE repositories (without it the `boot-sysctl` service fails in an nspawn container).

2) The "mkosi wrapper"
Given the recent state of things, sometimes systemd-dissect fails when manipulating the image with `Failed to dissect image '/home/runner/work/systemd/systemd/image.raw': Connection timed out`. There were some attempts to address this in the past (#18851, 48a3cf58d5ad9cf2a4a4e6167171070b88c25369., etc.), but the issue is still present (and probably related to https://github.com/systemd/systemd/issues/17469). So, until it's resolved, let's use a relatively simple wrapper to retry the mkosi command in case it dies with the aforementioned message.

After a couple of tries it looks like it's DTRT:
image

Also, it successfully detects #21671 when I revert the fix:
https://github.com/systemd-ci-incubator/systemd/runs/4473682100?check_suite_focus=true

If there's a way to make this even more simpler/nicer/cleaner, please let me know.

/cc @DaanDeMeyer (I'm not sure if the pexpect script was there with a specific purpose, other than shutting down the machine)

### CODE REVIEW

File: .github/workflows/run_mkosi.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```
Comment (bluca) [inline .github/workflows/run_mkosi.sh:1]: SPDX line

**Comment by keszybz** [comment]
LGTM but please add SPDX headers everywhere.

**Comment by mrc0mmand** [comment]
Just added the SPDX headers, upgrading the green label.

**Comment by DaanDeMeyer** [comment]
I've always entertained the idea of using D-Bus calls over SSH to retrieve information from the running VM/container instead of using services to run commands. Maybe something to look into for the future

e.g. `busctl -H <qemu-hostname> call <method-for-failed-services>`

pexpect was there to verify the machine actually got to a login prompt and was able to shutdown correctly

**Comment by mrc0mmand** [comment]
> I've always entertained the idea of using D-Bus calls over SSH to retrieve information from the running VM/container instead of using services to run commands. Maybe something to look into for the future
> 
> e.g. `busctl -H <qemu-hostname> call <method-for-failed-services>`
> 

That's definitely something we should look into in the future

> pexpect was there to verify the machine actually got to a login prompt and was able to shutdown correctly

I see, thanks.

**Pull request merged**